src/js/Content/Features/Community/Badges/FBadgeSortAndFilter.js

```diff
@@ -175,67 +175,80 @@ export default class FBadgeSortAndFilter extends Feature {
         this._hasAllPagesLoaded = true;
 
         const sheetNode = document.querySelector(".badges_sheet");
-
-        // let images = Viewport.getVariableFromDom("g_rgDelayedLoadImages", "object");
+        let images = HTMLParser.getVariableFromDom("g_rgDelayedLoadImages", "object");
 
         await this.context.eachBadgePage(dom => {
-            for (const node of dom.querySelectorAll(".badge_row")) {
-                sheetNode.append(node);
-            }
+            sheetNode.append(
+                ...Array.from(dom.querySelector(".badges_sheet").children).filter(row => !(row instanceof HTMLScriptElement))
+            );
 
             this.context.triggerCallbacks();
 
-            // images = Object.assign(images, Viewport.getVariableFromDom("g_rgDelayedLoadImages", "object", dom));
+            images = Object.assign(images, HTMLParser.getVariableFromDom("g_rgDelayedLoadImages", "object", dom));
         });
 
         for (const node of document.querySelectorAll(".profile_paging")) {
             node.style.display = "none";
         }
 
-        /*
-         * TODO this doesn't seem to work, can't figure out why right now. Lazy loader doesn't see updated object?
-         * ExtensionLayer.runInPageContext("function(){g_rgDelayedLoadImages = " + JSON.stringify(images) + ";}");
-         * resetLazyLoader();
-         */
+        Page.runInPageContext(images => {
+            const f = window.SteamFacade;
+
+            f.globalSet("g_rgDelayedLoadImages", images);
+
+            // Clear registered image lazy loader watchers
+            CScrollOffsetWatcher.sm_rgWatchers = []; // eslint-disable-line no-undef, camelcase
+
+            // Recreate image lazy loader watchers
+            for (const node of document.querySelectorAll("div[id^=image_group_scroll_badge_images_]")) {
+                f.loadImageGroupOnScroll(node.id, node.id.slice(19));
+            }
+        }, [images]);
     }
 
     _sortBadgeRows(activeText, nodeValueCallback) {
+        const sheetNode = document.querySelector(".badges_sheet");
+
+        // Remove script tags that'll get in the way of associating rows with their scroll elements
+        for (const script of sheetNode.querySelectorAll(":scope > script")) {
+            script.remove();
+        }
+
         const badgeRows = [];
         for (const node of document.querySelectorAll(".badge_row")) {
-            badgeRows.push([node.outerHTML, nodeValueCallback(node)]);
-            node.remove();
+            const scrollEl = node.previousElementSibling;
+            if (!scrollEl.id.startsWith("image_group_")) { continue; }
```

Comment:
If this check fails, it'll likely fail for all rows, which means Steam may have changed something and sorting will probably not work at all. Maybe `return` instead?